File: v1/src/main/java/com/api/v1/dtos/CustomerRegistrationRequestDto.java
package com.api.v1.dtos;

import java.time.LocalDate;

import com.api.v1.annotations.PhoneNumber;
import com.api.v1.annotations.SSN;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record CustomerRegistrationRequestDto(
        @NotBlank String firstName,
        String middleName,
        @NotBlank String lastName,
        @SSN String ssn,
        @NotNull LocalDate birthDate,
        @NotBlank @Email String email,
        @NotBlank String address,
        @PhoneNumber String phoneNumber
) {
}

File: v2/src/main/kotlin/com/api/v2/CustomerRegistrationRequestDto.kt
package com.api.v2

import java.time.LocalDate

data class CustomerRegistrationRequestDto(
    var firstName: String,
    var middleName: String,
    var lastName: String,
    var ssn: String,
    var birthDate: LocalDate,
    var email: String,
    var phoneNumber: String,
    var gender: String,
)